packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,353 @@
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { Download, Upload, IZosFilesResponse, IDownloadOptions, IUploadOptions } from "@zowe/zos-files-for-zowe-sdk";
+import { AbstractSession, IHandlerParameters, ImperativeError, ProcessUtils, GuiResult,
+    TextUtils, IDiffNameOptions, CliUtils } from "@zowe/imperative";
+import { CompareBaseHelper } from "../compare/CompareBaseHelper";
+import { existsSync, unlinkSync } from "fs";
+import { tmpdir } from "os";
+import path = require("path");
+import LocalfileDatasetHandler from "../compare/lf-ds/LocalfileDataset.handler";
+import LocalfileUssHandler from "../compare/lf-uss/LocalfileUss.handler";
+
+
+/**
+ * enum of prompts to be used as input to {@link EditUtilities.promptUser} during the file editing process
+ * @export
+ * @enum
+ */
+export enum Prompt {
+    useStash,
+    doneEditing,
+    continueToUpload,
+    viewUpdatedRemote
+}
+
+/**
+ * The possible types of file {@link ILocalFile}
+ * @export
+ * @type
+ */
+export type EditFileType = "uss" | "ds";
+
+/**
+ * A class to hold pertinent information about the local file during the editing process
+ * @export
+ * @interface
+ */
+export interface ILocalFile {
+    tempPath: string | null;
+    fileName: string;
+    fileType: EditFileType;
+    guiAvail: boolean;
+    zosResp: IZosFilesResponse | null;
+    encoding?: string | null;
+}
+
+/**
+ * A shared utility class that uss and ds handlers use for local file editing
+ * @export
+ * @class
+ */
+export class EditUtilities {
+    /**
+     * Builds a temp path where local file will be saved. If uss file, file name will be hashed
+     * to prevent any conflicts with file naming. A given filename will always result in the
+     * same unique file path.
+     * @param {ILocalFile} lfFile - object containing pertinent information about the local file during the editing process
+     * @returns {Promise<string>} - returns unique file path for temp file
+     * @memberof EditUtilities
+     */
+    public static async buildTempPath(lfFile: ILocalFile, commandParameters: IHandlerParameters): Promise<string>{
+        // find the appropriate extension for either uss or ds in two (long) operations
+        const ussExt = (lfFile.fileType === 'uss' && lfFile.fileName.includes(".")) ? lfFile.fileName.split(".").pop() : "";
+        const ext = "."  + (lfFile.fileType === 'uss' ? ussExt : (commandParameters.arguments.extension ?? "txt"));
+        if (lfFile.fileType === 'uss'){
+            // Hash in a repeatable way if uss fileName (incase there are special characters in name)
+            const crypto = require("crypto");
+            const hash = crypto.createHash('sha256').update(lfFile.fileName).digest('hex');
+
+            return path.join(tmpdir(), hash + ext);
```

Comment:
If a USS file has no extension, this will still append a period to the end of the filename of the temporary file. This prevents the user from being able to open the temporary file.